xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-api/src/main/java/org/xwiki/livedata/internal/StringLiveDataConfigurationResolver.java

```diff
@@ -253,6 +258,21 @@ private void normalizeIcon(ObjectNode descriptor, ObjectMapper objectMapper)
         } else if (!icon.isObject()) {
             descriptor.remove(ICON);
         }
+        normalizeIconClasses(descriptor);
+    }
+
+    private static void normalizeIconClasses(ObjectNode descriptor)
+    {
+        JsonNode icon = descriptor.path(ICON);
+        if (icon.isObject()) {
+            JsonNode extraClasses = descriptor.path(EXTRA_CLASSES);
+            if (extraClasses.isTextual() && icon.has(CSS_CLASS) && icon.path(CSS_CLASS).isTextual()) {
```

Comment:
The extraClass should definitely be there. I missed that because in practice, `cssClass` is always initialized as the empty string but the icon sets.
I'll cover that case anyway.
Thanks for catching this.

Reply:
Done